File: src/main/java/com/aries/tutorial/adapter/SystemEventAdapter.java
package com.aries.tutorial.adapter;

import com.aries.extension.data.SystemEventData;
import com.aries.extension.handler.SystemEventHandler;
import com.aries.extension.util.LogUtil;
import com.aries.extension.util.PropertyUtil;
import com.aries.tutorial.util.AdapterFormatter;

public class SystemEventAdapter implements SystemEventHandler {
    @Override
    public void on(SystemEventData[] events) {
        String subject = PropertyUtil.getValue("system_event_adapter", "subject", "Unknown subject");
        LogUtil.info("[SystemEventAdapter] - " + subject + " (events=" + events.length + ")");

        int idx = 0;
        for (SystemEventData data : events) {
            idx++;
            LogUtil.info(AdapterFormatter.formatSystemEvent(idx, data));
        }
    }
}

File: src/main/kotlin/com/aries/tutorial2/adapter/SystemEventAdapter.kt
package com.aries.tutorial2.adapter

import com.aries.extension.data.SystemEventData
import com.aries.extension.handler.SystemEventHandler
import com.aries.extension.util.LogUtil
import com.aries.extension.util.PropertyUtil
import com.aries.tutorial2.util.AdapterFormatter

class SystemEventAdapter : SystemEventHandler {
    override fun on(events: Array<SystemEventData>) {
        val subject = PropertyUtil.getValue("system_event_adapter", "subject", "Unknown subject")
        LogUtil.info("[SystemEventAdapter] - $subject (events=${events.size})")

        events.forEachIndexed { index, data ->
            LogUtil.info(AdapterFormatter.formatSystemEvent(index + 1, data))
        }
    }
}
